src/kind2.ml

```diff
@@ -96,22 +96,21 @@ let setup : unit -> any_input = fun () ->
       KEvent.terminate_log () ;
       exit ExitCodes.error
   with (* Could not create input system. *)
-  | LustreAst.Parser_error ->
-    (* Don't do anything for parser error, they should already have printed
-    some stuff. *)
-    KEvent.terminate_log () ;
-    exit ExitCodes.error
+  | LustreAst.Parser_error  ->
+     (* terminate log after printing file name and exit as the error is already on the log file *)
+     KEvent.log L_fatal "While parsing file %s" (Flags.input_file ()); 
```

Comment:
You should check whether the filename is empty (it is reading from standard input), and adjust the message accordingly. Same comment for lexer error.

Reply:
makes sense. fixed it now.
```
cat tests/lustre/syntax-errors/array-syntax.lus |  ./bin/kind2 -vv --only_parse true
 kind2 cf9613e-dirty

 Parsing standard input.

<Error> While parsing standard input
 Lexing error: Unrecognized token % (0x25) found at line 4, column 5
```
and while parsing from a file:

```
./bin/kind2 -vv --only_parse true tests/lustre/syntax-errors/array-syntax.lus
 kind2 cf9613e-dirty

 Parsing input file 'tests/lustre/syntax-errors/array-syntax.lus'.

<Error> While parsing input file 'tests/lustre/syntax-errors/array-syntax.lus'
 Lexing error: Unrecognized token % (0x25) found at line 4, column 5
```